recipes/logr/0.1.0/conanfile.py

```diff
@@ -1,97 +1,104 @@
+    options = {
+        "backend": ["spdlog", "glog", "log4cplus", "log4cplus-unicode", None],
+    }
+    default_options = {
+        "backend": "spdlog",
+    }
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return 17
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    def requirements(self):
-        self.requires("fmt/7.1.2")
-
-        if self.options.backend == "spdlog":
-            self.requires("spdlog/1.8.0")
-        elif self.options.backend == "glog":
-            self.requires("glog/0.4.0")
-        elif self.options.backend == "log4cplus":
-            self.requires("log4cplus/2.0.5")
-        elif self.options.backend == "log4cplus-unicode":
-            self.requires("log4cplus/2.0.5")
-
-    def configure(self):
-        minimal_cpp_standard = "17"
-        if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, minimal_cpp_standard)
-        minimal_version = {
+    def _compilers_minimum_version(self):
+        return {
             "gcc": "7",
             "clang": "7",
             "apple-clang": "10",
-            "Visual Studio": "16"
+            "Visual Studio": "16",
+            "msvc": "192",
         }
-        compiler = str(self.settings.compiler)
-        if compiler not in minimal_version:
-            self.output.warn(
-                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
-            self.output.warn(
-                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
-            return
-
-        version = tools.Version(self.settings.compiler.version)
-        if version < minimal_version[compiler]:
-            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
 
+    def export_sources(self):
+        export_conandata_patches(self)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["LOGR_WITH_SPDLOG_BACKEND"] = self.options.backend == "spdlog"
-        self._cmake.definitions["LOGR_WITH_GLOG_BACKEND"] = self.options.backend == "glog"
-        self._cmake.definitions["LOGR_WITH_LOG4CPLUS_BACKEND"] = self.options.backend in ["log4cplus", "log4cplus-unicode"]
-
-        self._cmake.definitions["LOGR_INSTALL"] = True
-
-        self._cmake.configure( build_folder=self._build_subfolder )
-        return self._cmake
-
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def requirements(self):
+        self.requires("fmt/9.1.0")
```

Comment:
Older version of logr using more modern fmt than newer versions of logr?

Reply:
Sorry I don't know details.
Because logr/0.1.0 can be build with spdlog/1.10.0, I upgrade fmt version for logr/0.1.0.

Reply:
spdlog is not able to use the latest fmt version, so we need to keep both fmt and spdlog aligned. I would consider something more dynamic if possible like:

```python
if self.options.backend == "spdlog":
    self.requires("spdlog/1.11.0")
    self.requires("fmt/9.1.0")
else:
    self.requires("fmt/10.0.0")
```

So we can use the latest or the older. Plus, `fmt` is a good case for version range adoption.

Reply:
@uilianries 
Thank you for your advice!
I fixed this recipe.